storage/dvo_recommendations_storage.go

```diff
@@ -411,6 +421,118 @@ func (storage DVORecommendationsDBStorage) getReportedAtMap(orgID types.OrgID, c
 	return reportedAtMap, err
 }
 
+// ReadWorkloadsForOrganization returns all rows from dvo.dvo_report table for given organizaiton
+func (storage DVORecommendationsDBStorage) ReadWorkloadsForOrganization(orgID types.OrgID) (
+	workloads []types.DVOReport,
+	err error,
+) {
+	tStart := time.Now()
+	query := `
+		SELECT cluster_id, namespace_id, namespace_name, recommendations, objects, reported_at, last_checked_at
+		FROM dvo.dvo_report
+		WHERE org_id = $1
+	`
+
+	// #nosec G202
+	rows, err := storage.connection.Query(query, orgID)
+
+	err = types.ConvertDBError(err, orgID)
+	if err != nil {
+		return workloads, err
+	}
+
+	defer closeRows(rows)
+
+	var (
+		dvoReport       types.DVOReport
+		lastCheckedAtDB sql.NullTime
+		reportedAtDB    sql.NullTime
+		count           uint
+	)
+	for rows.Next() {
+		err = rows.Scan(
+			&dvoReport.ClusterID,
+			&dvoReport.NamespaceID,
+			&dvoReport.NamespaceName,
+			&dvoReport.Recommendations,
+			&dvoReport.Objects,
+			&lastCheckedAtDB,
+			&reportedAtDB,
+		)
+		if err != nil {
+			log.Error().Err(err).Msg("ReadWorkloadsForOrganization")
+		}
+
+		// convert timestamps to string
+		dvoReport.LastCheckedAt = types.Timestamp(lastCheckedAtDB.Time.UTC().Format(time.RFC3339))
+		dvoReport.ReportedAt = types.Timestamp(reportedAtDB.Time.UTC().Format(time.RFC3339))
+
+		workloads = append(workloads, dvoReport)
+		count++
+	}
+
+	log.Debug().Int(orgIDStr, int(orgID)).Msgf("ReadWorkloadsForOrganization processed %d rows in %v", count, time.Since(tStart))
+
+	return workloads, err
+}
+
+// ReadWorkloadsForClusterAndNamespace returns a single result from the dvo.dvo_report table
```

Comment:
this function shares a lot of code with `ReadWorkloadsForOrganization`, but probably it's not worth it to refactor them given the benefit we would get